config/dispatcher/dispatcher.ex

```diff
@@ -1,532 +1,566 @@
 defmodule Dispatcher do
-  use Plug.Router
-
-  def start(_argv) do
-    port = 80
-    IO.puts "Starting Plug with Cowboy on port #{port}"
-    Plug.Adapters.Cowboy.http __MODULE__, [], port: port
-    :timer.sleep(:infinity)
-  end
-
-  plug Plug.Logger
-  plug :match
-  plug :dispatch
-
-  # In order to forward the 'themes' resource to the
-  # resource service, use the following forward rule.
-  #
-  # docker-compose stop; docker-compose rm; docker-compose up
-  # after altering this file.
-  #
-  # match "/themes/*path" do
-  #   Proxy.forward conn, path, "http://resource/themes/"
-  # end
-
-  match "/bestuurseenheden/*path" do
+  use Matcher
+
+  define_accept_types [
+    json: [ "application/json", "application/vnd.api+json" ],
+    html: [ "text/html", "application/xhtml+html" ],
+    sparql: [ "application/sparql-results+json" ],
+    any: [ "*/*" ]
+  ]
+
+  define_layers [ :static, :sparql, :api_services, :frontend, :resources, :not_found ]
+
+  options "/*path", _ do
+    conn
+    |> Plug.Conn.put_resp_header( "access-control-allow-headers", "content-type,accept" )
+    |> Plug.Conn.put_resp_header( "access-control-allow-methods", "*" )
+    |> send_resp( 200, "{ \"message\": \"ok\" }" )
+  end
+
+  match "/bestuurseenheden/*path", %{ layer: :resources, accept: %{ json: true } } do
```

Comment:
I see allot of code duplication in `%{ layer: :resources, accept: %{ json: true } }`. 

In the documentation I noticed the following:
```
@json %{ accept: %{ json: true } }

match "/*_path", @json do
... # further impl.
```
Couldn't we improve the code duplication by introducing our own variables:
```
@resource %{ layer: :resources, accept: %{ json: true } }
@json_service %{ layer: :api_services, accept: %{ json: true } }
@html_service %{ layer: :api_services, accept: %{ html: true } }
...
```

Reply:
Not super familiar with elixir but I would guess that you can indeed just define a variable and use that instead